### ISSUE TITLE
TensorFlow Training silently fails (seems like it never starts)

### ISSUE THREAD

**Issue by DorianBattivelli** [DeepLabCut/DeepLabCut#2812]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Bug description

Hello, 

I'm running a single animal project from DLC 2, installed on a container on a Linux machine. 
While everything works fine until the manual labeling step, the training does not work. I get no error message, it seems only that the training never starts. 
Since we didn't solve the problem in my previous post, could I send you a video to see if you manage to reproduce this issue?

Thank you, 
Best,

### Operating System

operating system Linux

### DeepLabCut version

dlc version
2

### DeepLabCut mode

single animal

### Device type

gpu

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/DeepLabCut/DeepLabCut/blob/master/CODE_OF_CONDUCT.md)

**Comment by DorianBattivelli** [comment]
Here again the resulting joblog, and where it gets stuck: 

INFO:    Environment variable SINGULARITY_BINDPATH is set, but APPTAINER_BINDPATH is preferred
INFO:    Using cached SIF image
INFO:    Environment variable SINGULARITYENV_TMPDIR is set, but APPTAINERENV_TMPDIR is preferred
ERROR: ld.so: object 'libdlfaker.so' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
ERROR: ld.so: object 'libvglfaker.so' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
ERROR: ld.so: object 'libdlfaker.so' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
ERROR: ld.so: object 'libvglfaker.so' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
2024-12-11 16:46:26.345970: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0
Config:
{'all_joints': [[0], [1], [2], [3], [4], [5], [6], [7]],
 'all_joints_names': ['Ear_left',
                      'Ear_right',
                      'Nose',
                      'Center',
                      'Lateral_left',
                      'Lateral_right',
                      'Tail_base',
                      'Tail_end'],
 'alpha_r': 0.02,
 'apply_prob': 0.5,
 'batch_size': 1,
 'clahe': True,
 'claheratio': 0.1,
 'crop_pad': 0,
 'crop_sampling': 'hybrid',
 'crop_size': [400, 400],
 'cropratio': 0.4,
 'dataset': 'training-datasets/iteration-0/UnaugmentedDataSet_EncodingDec11/Encoding_DBattivelli95shuffle1.mat',
 'dataset_type': 'imgaug',
 'decay_steps': 30000,
 'deterministic': False,
 'display_iters': 1000,
 'edge': False,
 'emboss': {'alpha': [0.0, 1.0], 'embossratio': 0.1, 'strength': [0.5, 1.5]},
 'fg_fraction': 0.25,
 'global_scale': 0.8,
 'histeq': True,
 'histeqratio': 0.1,
 'init_weights': '/usr/local/lib/python3.8/dist-packages/deeplabcut/pose_estimation_tensorflow/models/pretrained/resnet_v1_50.ckpt',
 'intermediate_supervision': False,
 'intermediate_supervision_layer': 12,
 'location_refinement': True,
 'locref_huber_loss': True,
 'locref_loss_weight': 0.05,
 'locref_stdev': 7.2801,
 'log_dir': 'log',
 'lr_init': 0.0005,
 'max_input_size': 1500,
 'max_shift': 0.4,
 'mean_pixel': [123.68, 116.779, 103.939],
 'metadataset': 'training-datasets/iteration-0/UnaugmentedDataSet_EncodingDec11/Documentation_data-Encoding_95shuffle1.pickle',
 'min_input_size': 64,
 'mirror': False,
 'multi_stage': False,
 'multi_step': [[0.005, 10000],
                [0.02, 430000],
                [0.002, 730000],
                [0.001, 1030000]],
 'net_type': 'resnet_50',
 'num_joints': 8,
 'optimizer': 'sgd',
 'pairwise_huber_loss': False,
 'pairwise_predict': False,
 'partaffinityfield_predict': False,
 'pos_dist_thresh': 17,
 'pre_resize': [],
 'project_path': '/g/gross/Dorian2/Encoding-DBattivelli-2024-12-11',
 'regularize': False,
 'rotation': 25,
 'rotratio': 0.4,
 'save_iters': 50000,
 'scale_jitter_lo': 0.5,
 'scale_jitter_up': 1.25,
 'scoremap_dir': 'test',
 'sharpen': False,
 'sharpenratio': 0.3,
 'shuffle': True,
 'snapshot_prefix': '/g/gross/Dorian2/Encoding-DBattivelli-2024-12-11/dlc-models/iteration-0/EncodingDec11-trainset95shuffle1/train/snapshot',
 'stride': 8.0,
 'weigh_negatives': False,
 'weigh_only_present_joints': False,
 'weigh_part_predictions': False,
 'weight_decay': 0.0001}
2024-12-11 16:46:35.546765: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcuda.so.1
2024-12-11 16:46:35.777270: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1733] Found device 0 with properties: 
pciBusID: 0000:01:00.0 name: NVIDIA A100-PCIE-40GB computeCapability: 8.0
coreClock: 1.41GHz coreCount: 108 deviceMemorySize: 39.50GiB deviceMemoryBandwidth: 1.41TiB/s
2024-12-11 16:46:35.777293: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0
2024-12-11 16:46:35.879831: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublas.so.11
2024-12-11 16:46:35.879886: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublasLt.so.11
2024-12-11 16:46:35.929401: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcufft.so.10
2024-12-11 16:46:35.990031: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcurand.so.10
2024-12-11 16:46:36.004352: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcusolver.so.11
2024-12-11 16:46:36.026723: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcusparse.so.11
2024-12-11 16:46:36.027937: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudnn.so.8
2024-12-11 16:46:36.032077: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1871] Adding visible gpu devices: 0
2024-12-11 16:46:36.032719: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-12-11 16:46:36.036836: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1733] Found device 0 with properties: 
pciBusID: 0000:01:00.0 name: NVIDIA A100-PCIE-40GB computeCapability: 8.0
coreClock: 1.41GHz coreCount: 108 deviceMemorySize: 39.50GiB deviceMemoryBandwidth: 1.41TiB/s
2024-12-11 16:46:36.040709: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1871] Adding visible gpu devices: 0
2024-12-11 16:46:36.042835: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0
2024-12-11 16:46:39.717211: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1258] Device interconnect StreamExecutor with strength 1 edge matrix:
2024-12-11 16:46:39.717257: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1264]      0 
2024-12-11 16:46:39.717264: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1277] 0:   N 
2024-12-11 16:46:39.722771: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1418] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 38335 MB memory) -> physical GPU (device: 0, name: NVIDIA A100-PCIE-40GB, pci bus id: 0000:01:00.0, compute capability: 8.0)
2024-12-11 16:46:40.019843: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1733] Found device 0 with properties: 
pciBusID: 0000:01:00.0 name: NVIDIA A100-PCIE-40GB computeCapability: 8.0
coreClock: 1.41GHz coreCount: 108 deviceMemorySize: 39.50GiB deviceMemoryBandwidth: 1.41TiB/s
2024-12-11 16:46:40.023184: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1871] Adding visible gpu devices: 0
2024-12-11 16:46:40.023210: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1258] Device interconnect StreamExecutor with strength 1 edge matrix:
2024-12-11 16:46:40.023215: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1264]      0 
2024-12-11 16:46:40.023218: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1277] 0:   N 
2024-12-11 16:46:40.026529: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1418] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 38335 MB memory) -> physical GPU (device: 0, name: NVIDIA A100-PCIE-40GB, pci bus id: 0000:01:00.0, compute capability: 8.0)
2024-12-11 16:50:43.387666: I tensorflow/core/platform/profile_utils/cpu_utils.cc:114] CPU Frequency: 2000080000 Hz

**Comment by DorianBattivelli** [comment]
I actually found that the problem is not the format but the dimension of the image : 2020.2052
My previous videos on which I successfully run training were 2020.2034. 
If I convert the new video so that they are 1060.1080, it works. 

Is that possible that training fails on too big format? 
I tried to reduce the batchsize to 1 but it does solve anything. 
Thank you

**Comment by n-poulsen** [comment]
Hi @DorianBattivelli - yes that might be the issue (I'm wondering if your GPU might be silently failing if it does not have enough memory to fit such large frames). 

What do you have set as the `max_input_size` in the `pose_cfg.yaml` for the shuffle for which you're training?

**Comment by DorianBattivelli** [comment]
max_input_size: 1500

**Comment by n-poulsen** [comment]
This might be where the issue comes from: if you set `max_input_size: 2500`, does your model train at all (make sure you set `displayiters: 1` as well so you get )?

Note that you should not train with images that are this large (see our [docs for `max_input_size`](https://deeplabcut.github.io/DeepLabCut/docs/recipes/pose_cfg_file_breakdown.html#a-max-input-size-and-min-input-size) and the critical point when [selecting frames to label](https://deeplabcut.github.io/DeepLabCut/docs/maDLC_UserGuide.html#select-frames-to-label)).

**Comment by DorianBattivelli** [comment]
Amazing, it worked even with 2000. 
Thanks a lot :)
